src/mbedtls_context.cpp

```diff
@@ -673,32 +674,46 @@ namespace sockpp {
     }
 
 
-    // callback from mbedTLS cert validation (see above)
+    // Callback from mbedTLS cert validation (see above)
+    //
+    // When a pinned cert is specified, the verify_callback will compare the pinned cert with
+    // each cert in the chain. If one of the certs in the chain matches with the pinned cert,
+    // the server servers are trusted.
+    //
+    // The verify_callback is called for each cert in the chain from root to leaf cert. The
+    // pinned_cert_validation_result_ will store the previous comparison result. If the
+    // comparison result is already matched, the comparison will be skipped.
+    //
+    // The last callback for the leaf cert is where the comparison or verification is set
+    // to the status flags. The flags of the parent certs are ignored (clear).
+    //
     int mbedtls_context::verify_callback(mbedtls_x509_crt *crt, int depth, uint32_t *flags) {
-        if (depth != 0) {
-            if(pinned_cert_) {
-                // We only care that the end cert matches, clear all other errors
-                *flags = 0;
-            }
-            
-            return 0;
+        if (pinned_cert_ && pinned_cert_validation_result_ == 0) {
+            pinned_cert_validation_result_ = (crt->raw.len == pinned_cert_->raw.len &&
```

Comment:
appears to be a bool.

Reply:
The original code stores the comparison result in int but yes, bool is more correct. I will change it to bool.